src/browse.vue

```diff
@@ -509,6 +515,25 @@ export default {
 
             return params;
         },
+
+        processTableFields(endpointFields) {
+            const newFields = [];
+            this.extraFields.forEach((fieldDefinition, index) => {
+                // find field to replace the render
+                const fieldIndex = endpointFields.findIndex(field => [fieldDefinition.name, fieldDefinition.field].includes(field.name));
+                if (fieldIndex != -1) {
+                    const fieldName = endpointFields[fieldIndex].name;
+                    endpointFields[fieldIndex] = { ...endpointFields[fieldIndex], ...fieldDefinition, fieldName };
+                } else {
+                    newFields.push(index);
+                }
+            });
+            newFields.forEach((fieldIndex) => {
```

Comment:
@SparoHawk 
In response to your last comment. Why iterate again it is because the function we make two things.

We can redefine a custom render to a field that already exist in backend. or add a field(at the end).

Reply:
This is an example of what I mean:

```js
        processTableFields(endpointFields) {
            const newFields = [];
            this.extraFields.forEach((fieldDefinition, index) => {
                // find field to replace the render
                const fieldIndex = endpointFields.findIndex(field => [fieldDefinition.name, fieldDefinition.field].includes(field.name));
                if (fieldIndex != -1) {
                    const fieldName = endpointFields[fieldIndex].name;
                    endpointFields[fieldIndex] = { ...endpointFields[fieldIndex], ...fieldDefinition, fieldName };
                } else {
					endpointFields.push(this.extraFields[index]);
                }
            });

            return endpointFields;
        },
```

Reply:
Yep, you'll right.

I'll make the change